pandas/core/indexes/multi.py

```diff
@@ -975,7 +975,7 @@ def _try_mi(k):
         except KeyError as e1:
             try:
                 return _try_mi(key)
-            except KeyError:
+            except (KeyError, IndexError, ValueError):
```

Comment:
Can you clarify this change? It feels like we just wanted to ignore `KeyError` exceptions, and I'm sure you have a reason to also ignore `IndexError` and `ValueError`, but it's not obvious to me.